instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt

```diff
@@ -247,6 +249,32 @@ class RouteRefreshTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.ja
         waitForRouteToSuccessfullyRefresh()
     }
 
+    @Test
+    fun routeAlternativeMetadataUpdatedAlongWithRouteRefresh() = sdkTest {
+        val routeOptions = generateRouteOptions(coordinates)
+        val routes = mapboxNavigation.requestRoutes(routeOptions)
+            .getSuccessfulResultOrThrowException()
+            .routes
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(routes)
+        val alternativesMetadataLegacy = mapboxNavigation.getAlternativeMetadataFor(routes).first()
+        mapboxNavigation.startTripSession()
+        stayOnInitialPosition()
+        mapboxNavigation.routesUpdates()
+            .first { it.reason == ROUTES_UPDATE_REASON_REFRESH }
+
+        val alternativesMetadata = mapboxNavigation.getAlternativeMetadataFor(
+            mapboxNavigation.getNavigationRoutes()
+        ).first()
+
+        assertNotNull(alternativesMetadataLegacy)
+        assertNotNull(alternativesMetadata)
+        assertEquals(
+            alternativesMetadataLegacy.navigationRoute.id,
+            alternativesMetadata.navigationRoute.id
+        )
+        assertNotEquals(alternativesMetadataLegacy, alternativesMetadata)
```

Comment:
this test confirms that metadata is updated after refresh. Should this be enough? 🤔

Reply:
We're mocking the requests so we could be more strict in assertions to make sure that data is correct as well, not only that it's different.